dsl/src/main/scala/org/http4s/dsl/impl/Path.scala

```diff
@@ -166,12 +166,21 @@ object /: {
     }
 }
 
-protected class PathVar[A](cast: String => Try[A]) {
+/** Abstract extractor of a path variable:
+  * {{{
+  *   enum Color:
+  *     case Red, Green, Blue
+  *
+  *   val ColorPath = new PathVar(str => Try(Color.valueOf(str)))
+  *
+  *   Path("/Green") match {
+  *     case Root / ColorPath(color) => ...
+  * }}}
+  */
+class PathVar[A](cast: String => Try[A]) {
   def unapply(str: String): Option[A] =
-    if (!str.isEmpty)
-      cast(str).toOption
-    else
-      None
+    if (str.nonEmpty) cast(str).toOption
+    else None
```

Comment:
Wild thoughts, I'm not an expert. I wonder if we can leave the class protected and instead expose smart constructors in a companion object, like
`PathVar.fromTry[A](f:String => Try[A])` or `PathVar.pure[A](f:String => A)` and similars

Reply:
That could be a good approach if the intent is to avoid inheritance

Another option might be to refactor to a SAM trait.

I would be more inclined to go the SAM trait route though.

Reply:
I would tend to agree with @TonioGela here, I would argue something using `Try` outside of a `fromTry` method does not feel "http4s-y" nor "Typelevel-y". Since the end result is creating an extractor, I would have assumed it exposes an `Option` internally

Reply:
Will adjust the code later today, thanks for the review/comments :3

Reply:
Thanks for the review and apologies for not commenting in a week. Making a first attempt at exposing and using `fromTry`, `fromPartialFunction`, and `pure`. (And making the class constructor private)